simplehub.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package pubsub
+
+import (
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+)
+
+// NewSimpleHub returns a new Hub instance.
+//
+// A simple hub does not touch the data that is passed through to Publish.
+// This data is passed through to each Subscriber. Note that all subscribers
+// are notified in parallel, and that no modification should be done to the
+// data or data races will occur.
+//
+// All handler functions passed into Subscribe methods of a SimpleHub should
+// be `func(Topic, interface{})`. The topic of the published method is the first
+// parameter, and the published data is the seconnd parameter.
+func NewSimpleHub() Hub {
+	return &simplehub{
+		logger: loggo.GetLogger("pubsub.simple"),
```

Comment:
It might be nice to parameterise the logger, in case we want to have multiple hubs in the same process.